dlt/extract/hints.py

```diff
@@ -332,12 +336,14 @@ def apply_hints(
                 else:
                     t.pop("schema_contract", None)
             if additional_table_hints is not None:
-                for k, v in additional_table_hints.items():
-                    if v:
-                        t[k] = v  # type: ignore[literal-required]
+                if additional_table_hints:
+                    if t.get("additional_table_hints") is not None:
+                        for k, v in additional_table_hints.items():
+                            t["additional_table_hints"][k] = v
                     else:
-                        t.pop(k, None)  # type: ignore[misc]
-                t.pop("additional_table_hints", None)  # type: ignore
+                        t["additional_table_hints"] = additional_table_hints
+                else:
```

Comment:
So if additional_table_hints is set to {} then they are fully popped and if there is at least one item, then it gets merged into the existing values? is this the correct behavior?

Reply:
yes, all hints behave like this.